### PR TITLE
Show-screenpipe-shortcut

### PR DESCRIPTION
**Pull request by EzraEllette**
---
name: add screenpipe overlay shortcut
about: submit changes to the project
title: "[pr] "
labels: ''
assignees: ''

---
/claim #513 

## description
brief description of the changes in this pr.

related issue: #

## type of change
- [x] bug fix
- [x] new feature
- [ ] breaking change
- [ ] documentation update

## how to test
1. 
2. 
3. 

## checklist
- [ ] i have read the [CONTRIBUTING.md](https://github.com/mediar-ai/screenpipe/blob/main/CONTRIBUTING.md) file 
- [ ] i have added the custom cursor AI prompt to my settings as mentioned in CONTRIBUTING.md and used to write this PR
- [ ] my code follows the project's style guidelines
- [ ] i have performed a self-review of my code
- [ ] i have updated the documentation if necessary
- [ ] my changes generate no new warnings
- [ ] i have added tests that prove my fix is effective or that my feature works
- [ ] all tests pass locally with my changes

## additional notes
any other relevant information about the pr.

### CODE REVIEW

File: screenpipe-app-tauri/src-tauri/src/main.rs [hunk 1]
```diff
@@ -177,6 +180,29 @@ async fn main() {
                 let _ = File::create(path.clone()).unwrap();
             }
 
+            // Setup Shortcuts
+            use tauri_plugin_global_shortcut::{
+                Code, GlobalShortcutExt, Modifiers, Shortcut, ShortcutState,
+            };
+
+            let show_window_shortcut =
+                Shortcut::new(Some(Modifiers::META.union(Modifiers::SHIFT)), Code::KeyZ);
```
[the remark this answers is not in the document]
Comment (EzraEllette) [reply screenpipe-app-tauri/src-tauri/src/main.rs:189]: It's difficult

Doing it now...

**Comment by EzraEllette** [comment]
@louis030195 Mostly finished. need to figure out activation policy settings. Looking into issues you mentioned

**Comment by louis030195** [comment]
@tribhuwan-kumar curious how it behave on windows?

few (minor) weird behaviour on macos:
- screenpipe on top of cursor (full screen cursor) when i click on minus it does not reduce (but cross does)
- setting full screen only set 80% height somehow 
- i expected to press key and it would fill the input in settings (dunno how to impl this UX, can be done later version)
- sometimes cannot move the window until pressing shortcut again

issues (i can fix before release):
- crash when using special character on shortcut (just use validator in ui input)


overall i am ready to merge, just small notes, does not have to be perfect

**Comment by tribhuwan-kumar** [comment]
> @tribhuwan-kumar curious how it behave on windows?
> few (minor) weird behaviour on macos:
>   * screenpipe on top of cursor (full screen cursor) when i click on minus it does not reduce (but cross does)
>   * setting full screen only set 80% height somehow
>   * i expected to press key and it would fill the input in settings (dunno how to impl this UX, can be done later version)
>   * sometimes cannot move the window until pressing shortcut again
> issues (i can fix before release):
>   * crash when using special character on shortcut (just use validator in ui input)
> overall i am ready to merge, just small notes, does not have to be perfect

let me test it on windows

**Comment by tribhuwan-kumar** [comment]
i merge it locally with latest main branch and tested it, got some error about `ActivationPolicy` in windows. [test.log](https://github.com/user-attachments/files/17449062/test.log)

**Comment by louis030195** [comment]
/tip $50 @tribhuwan-kumar 


thx for testing

**Pull request merged**